src/components/graph/util/model-functions.js

```diff
@@ -30,29 +30,9 @@ export function getFirstNodeOfType(elements, nodeType, buildStatus) {
     }
 }
 
-export function isNodeValid(workingNode, selectedNode) {
-    const readOnlyNode = workingNode?.readOnly;
-    const builtWorkingNode = workingNode?.buildStatus === 'BUILT';
-    const workingNodeIsSelectedNode =
-        selectedNode === null ||
-        selectedNode === undefined ||
-        workingNode?.id === selectedNode?.id;
+export function isNodeValid(currentNode) {
+    const readOnlyNode = currentNode?.data?.readOnly;
```

Comment:
is a read only node always valid or is this true for root ?

Reply:
Why readOnly ?
just test type != ROOT ?

Reply:
Just  `currentNode?.data?.type == ROOT || currentNode?.data?.buildStatus === 'BUILT'`

Reply:
good question this check is for all nodes